cli/pkg/kctrl/local/sources/vendir_runner.go

```diff
@@ -0,0 +1,110 @@
+package sources
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	goexec "os/exec"
+
+	cmdcore "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/core"
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/exec"
+)
+
+type VendirRunner struct {
+	ui cmdcore.AuthoringUI
+}
+
+func NewVendirRunner(ui cmdcore.AuthoringUI) VendirRunner {
+	return VendirRunner{ui: ui}
+}
+
+func (r VendirRunner) RunSync() error {
+	_, err := os.Stat(vendirFileName)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	if os.IsNotExist(err) {
+		return nil
+	}
```

Comment:
I guess sometimes it's okay to have nested `if`s
```suggestion
	if err != nil {
	        if os.IsNotExist(err) {
		       return nil
	        }
		return err
	}
```
BTW, what other kind of errors are we expecting here?

Reply:
Most I can think of is lack of permissions